components/PlayerDetailsList/PlayerDetailsList.tsx

```diff
@@ -37,54 +37,50 @@ function PlayerDetailsList(props: Props) {
     let forceUpdate = useForceUpdate()
     let router = useRouter()
 
-    let [listElements, setListElements] = useState<(Auction | BidForList)[]>(props.auctions || [])
-    let [allElementsLoaded, setAllElementsLoaded] = useState(props.auctions ? props.auctions.length < 12 : false)
-    let [playerName, setPlayerName] = useState('')
+    let [listElements, setListElements] = useState<(Auction | BidForList)[]>([])
+    let [allElementsLoaded, setAllElementsLoaded] = useState<boolean>()
+    let [playerName, setPlayerName] = useState<string>('')
     let isLoadingElements = useRef(false)
 
     useEffect(() => {
+
         mounted = true
         router.events.on('routeChangeStart', onRouteChange)
 
+        setPlayerName(props.playerName)
+
         let listState = getListState()
-        if (listState !== undefined) {
+        if (listState) {
             setListElements(listState.listElements)
             setAllElementsLoaded(listState.allElementsLoaded)
-            setTimeout(() => {
-                if (!mounted) {
-                    return
-                }
-                window.scrollTo({
-                    left: 0,
-                    top: listState!.yOffset,
-                    behavior: 'auto'
-                })
-            }, 100)
+            isLoadingElements.current = false
+        }
+        else {
+            // We need to set reset to true to remove the previous player's listElements
+            loadNewElements(true)
         }
 
+        setTimeout(() => {
+            if (!mounted) {
+                return
+            }
+            window.scrollTo({
+                left: 0,
+                top: listState!.yOffset,
+                behavior: 'auto'
+            })
+        }, 100)
+
```

Comment:
Why did you move this outside the if?
It now throws a exception, because `listState` is undefined and we try to read ``yOffset` of undefined
![image](https://user-images.githubusercontent.com/58751503/182408835-7d43d79e-9d08-42f0-a892-50c7c615442a.png)

Reply:
This a mistake on my end. Yes, this section of code should be placed inside of the `if`